hasjob/models/filterset.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+
+from flask import url_for
+from sqlalchemy.dialects import postgresql
+from sqlalchemy import event, DDL
+from . import db, BaseScopedNameMixin, JobType, JobCategory, Tag, Domain, Board
+from ..extapi import location_geodata
+
+
+__all__ = ['FilterSet']
+
+
+filterset_jobtype_table = db.Table('filterset_jobtype', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('jobtype_id', None, db.ForeignKey('jobtype.id'), primary_key=True, index=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+filterset_jobcategory_table = db.Table('filterset_jobcategory', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('jobcategory_id', None, db.ForeignKey('jobcategory.id'), primary_key=True, index=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+filterset_tag_table = db.Table('filterset_tag', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('tag_id', None, db.ForeignKey('tag.id'), primary_key=True, index=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+filterset_domain_table = db.Table('filterset_domain', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('domain_id', None, db.ForeignKey('domain.id'), primary_key=True, index=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+class FilterSet(BaseScopedNameMixin, db.Model):
+    """
+    Store filters to display a filtered set of jobs scoped by a board on SEO friendly URLs
+
+    Eg: `https://hasjob.co/f/machine-learning-jobs-in-bangalore`
+    """
+
+    __tablename__ = 'filterset'
+
+    board_id = db.Column(None, db.ForeignKey('board.id'), nullable=False, index=True)
+    board = db.relationship(Board)
+    parent = db.synonym('board')
+
+    #: Welcome text
+    description = db.Column(db.UnicodeText, nullable=False, default=u'')
+
+    #: Associated job types
+    types = db.relationship(JobType, secondary=filterset_jobtype_table)
+    #: Associated job categories
+    categories = db.relationship(JobCategory, secondary=filterset_jobcategory_table)
+    tags = db.relationship(Tag, secondary=filterset_tag_table)
+    domains = db.relationship(Domain, secondary=filterset_domain_table)
+    location_geonameids = db.Column(postgresql.ARRAY(db.Integer(), dimensions=1), default=[], nullable=False, index=True)
```

Comment:
No index here since there's a trigger below.

Reply:
Also, couldn't this field be just `geonameids`? That's the name of the relationship in other models, since we're not using ARRAY there (we should migrate).